docs/content/sips/xxx-trigger-executors.md

```diff
@@ -0,0 +1,65 @@
+title = "SIP xxx - Trigger Executors"
+template = "main"
+
+---
+
+Summary: A new architecture for Spin trigger executors.
+
+Owner: lann.martin@fermyon.com
+
+Created: April 29, 2022
+
+<!-- Updated: April 4, 2022 -->
+
+## Background
+
+Spin currently includes built-in implementations for two trigger types: HTTP, and Redis (PubSub).
+We want to allow for additional trigger types in the future, whether they are "internal" types developed and
+supported within the Spin codebase, or "external" types developed by the broader community.
+
+While it is currently possible to build external trigger types - as in the
+[TimerTrigger example](https://github.com/fermyon/spin/blob/main/examples/spin-timer/src/main.rs) - it requires replicating much of the
+functionality provided by the `spin up` command.
+
+## Proposal
+
+### Trigger types are implemented as "trigger executors"
+
+These executors are regular subcommands/binaries, but are not normally executed directly by
+users.
+
+* "Internal" triggers use `spin` subcommands, e.g. `spin trigger http`
+* "External" triggers use external binaries, e.g. `spin-trigger-timer`
+* Trigger executors share common code, including common CLI options in a `spin-trigger` crate
+
+### The `spin up` command changes
+
+It remains the primary entrypoint for starting a Spin app, but its behavior is changed to:
+
+* Detect the trigger type for an application
+* Find a trigger executor for that type (e.g. check the `PATH` for `"spin-trigger-${type}"`)
```

Comment:
What are the ramifications of different strategies here?  E.g. also checking current directory, or directory that contains `spin` - might improve convenience but are there any safety issues?

Reply:
I couldn't think of any safety issues as long as the prefix (`spin-trigger-`) doesnt collide with other projects, but I think it's something that a prototype would help clarify.

Reply:
So far we managed to dodge a _Spin_ configuration file (not the app manifest), but we might start have enough potential things to configure that we might need one.

One of the things would be setting the paths to the executables for triggers.
(other configurations could include template defaults, Wasmtime cache configuration..)